### PR TITLE
Support delegate registration - Closes #22

### LINKED ISSUE #22: Add support for delegate registration
Lisk Nano currently does not support registration of new delegates. We should add it.

### PR DESCRIPTION
**Pull request by alihaghighatkhah**
Creates a delegate registration directive which bind an event listener to a menu element and shows a modal containing a form for registering as delegate.

Closes #22

### CODE REVIEW

File: src/app/components/delegate-registration/delegateRegistration.pug [hunk 1]
```diff
@@ -0,0 +1,23 @@
+md-dialog.dialog-delegate-registration(aria-label='Transfering', style='width: 700px; max-width: 700px')
+  md-card
+    md-card-title
+      md-card-title-text
+        span.md-title Send
```
Comment (slaweet) [inline src/app/components/delegate-registration/delegateRegistration.pug]: The title should say "Delegate Registration", not "Send"

File: src/app/components/delegate-registration/delegateRegistration.pug [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    md-card-content
+      form(name='delegateRegistrationForm', ng-submit='$ctrl.form.onSubmit(delegateRegistrationForm)')
+        md-content(flex='100', flex-gt-sm='70', flex-offset-gt-sm='15')
+          md-input-container.md-block
+            label Delegate name
+            input(type='text', name='delegateName', ng-model='$ctrl.form.name', required, ng-disabled='$ctrl.loading')
+            div(ng-messages='delegateRegistrationForm.name.$error')
+              div(ng-message='required') Required
+              //- div(ng-message='pattern') Invalid
```
Comment (slaweet) [inline src/app/components/delegate-registration/delegateRegistration.pug]: Remove the commented out line

File: src/app/components/delegate-registration/delegateRegistration.pug [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+            md-input-container.md-block
+              div.fee Fee: {{$ctrl.form.fee}} LSK
+            md-input-container.md-block
+              p Becoming a delegate requires registration. You may choose your own delegate name, which can be used to promote your delegate. Only the top 101 delegates are eligible to forge. All fees are shared equally between the top 101 delegates.
```
Comment (slaweet) [inline src/app/components/delegate-registration/delegateRegistration.pug]: I would make this look more like the info block in vote dialog [0], that is: use "info" icon and separate by md-divider:
[0] https://github.com/LiskHQ/lisk-nano/blob/development/src/app/components/delegates/vote.pug#L22-L30

File: src/app/components/delegate-registration/delegateRegistration.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      name: '',
+      fee: 25,
+      error: '',
+      onSubmit: (form) => {
+        if (form.$valid) {
+          $peers.active.registerDelegate($scope.form.name.toLowerCase(), Account.get().passphrase)
+            .then(() => {
+              $scope.reset(form);
+
+              success.dialog({ text: 'Account was successfully registered as delegate.' })
```
Comment (slaweet) [inline src/app/components/delegate-registration/delegateRegistration.js]: This message should rather say that the registration is pending and will be processed with the next block.

File: src/app/components/delegate-registration/delegateRegistration.pug (hunk 3 above)
Comment (alihaghighatkhah) [inline src/app/components/delegate-registration/delegateRegistration.pug]: Done.

File: src/app/components/delegate-registration/delegateRegistration.pug (hunk 2 above)
Comment (alihaghighatkhah) [inline src/app/components/delegate-registration/delegateRegistration.pug]: Done

File: src/app/components/delegate-registration/delegateRegistration.pug (hunk 1 above)
Comment (alihaghighatkhah) [inline src/app/components/delegate-registration/delegateRegistration.pug]: Done

**Pull request merged**